admin_guide/high_availability.adoc

```diff
@@ -11,37 +11,53 @@
-impose operational requirements to handle failure cases when one or more router
-instances are "unreachable".
+Ipfailover manages a pool of Virtual IP (VIP) addresses on a set of nodes. Every VIP in the set will be serviced by a node selected from the set. As long a single node is available, the VIPs will be served. There is no way to explicitly distribute the VIPs over the nodes. so there may be nodes with no VIPs and other nodes with many VIPs. If there is only one node, all VIPs will be on it.
 
-For some IP-based traffic services, virtual IP addresses (VIPs) should always be
-serviced for as long as a single instance is available. This simplifies the
-operational overhead and handles failure cases gracefully.
+[NOTE]
+====
+The VIPs must be routable from outside the cluster.
+====
+
+Ipfailover monitors a port on each VIP to determine whether the port is reachable on the node. If the port is not reachable, the VIP will not be assigned to the node. If the port is set to "0", this check is suppressed.
+Ipfailover uses link:http://www.keepalived.org/[*Keepalived*] to host a set of externally accessible VIP addresses on a set of hosts. Each VIP is only serviced by a single host at a time. *Keepalived* uses the VRRP protocol to determine which host (from the set of hosts) will service which VIP. If a host becomes unavailable or if the service that *Keepalived* is watching does not respond, the VIP is switched to another host from the set. Thus, a VIP is always serviced as long as a host is available.
+
+{product-title} supports creation of ipfailover deployment configration, by running the `oadm ipfailover` command. The ipfailover deployment configration specifies the set of VIP addresses, and the set of nodes on which to service them. A cluster can have multiple ipfailover deployment configurations, with each managing its own set of unique VIP addresses. Each node in the ipfailover configuration runs an ipfailover pod, and this pod runs *Keepalived*.
+
+When using VIPs to access a pod with host networking (_e.g. a router_), the application pod should be running on all nodes that are running the ipfailover pods. This enables any of the ipfailover nodes to become the master and service the VIPs when needed. If application pods are not running on all nodes with ipfailover, either some ipfailover nodes will never service the VIPs or some application pods will never receive any traffic. Use the same selector and replication count, for both ipfailover and the application pods, to avoid this mismatch.
 
-IMPORTANT: Even though
-a service is highly available, performance can still be affected.
+While using VIPs to access a service, any of the nodes can be in the ipfailover set of nodes, since the service is reachable on all nodes (no matter where the application pod is running). Any of the ipfailover nodes can become master at any time. The service can either use externalIPs and a service port or it can use a nodePort.
 
```

Comment:
I think nodePort is a system term, found multiple instances in the docs https://docs.openshift.com/search.html?query=nodePort